### ISSUE TITLE
Breaking change of styles' loading order @ 0.6.0

### ISSUE THREAD

**Issue by keann** [bem/webpack-bem-loader#64]
Hi guys.

Let's say there is a following case:
```
├─ blocks
│ └─ Example
│ │ ├─ _mod1
│ │ │ └─ Example_mod1.css  /* styles #1 */
│ │ ├─ _mod2
│ │ │ └─ Example_mod2.css  /* styles #2 */
│ │ ├─ Example.css         /* styles #0 */
│ │ └─ Example.js
└─ app.js

/*   Example.js   */
...
import 'm:mod1';
export default decl({
  block: 'Example'
});

/*   app.js   */
...
import Example from 'b:Example m:mod2';
...
```
where _mod1_ is pretty lightweight and widely used (e.g. some _disabled_ state with one-two loc), thus it's just easier to include it as a base block functionality (but still keep in a separate folder for clarity). Obviously, as _mod1_ overrides block's base styles, its css should be loaded after the main one.

With loader version <=0.5.1 loading order of styles was predictable & ok: 0 -> 2 -> 1.
But with 0.6.0 now it is 1 -> 0 -> 2 and totally breaks existing logic, described above.

**Comment by Yeti-or** [comment]
Valid issue, we will fix it

**Comment by Yeti-or** [comment]
@keann check `v0.6.1`, please

**Comment by keann** [comment]
@Yeti-or @bem/import-notation@1.1.3 broke everything everywhere)

For example, this code
```javascript
/* project/blocks/Header/Header.js */
import pt from 'prop-types';
import { decl } from 'bem-react-core';

import 'm:font=roboto|ubuntu';
import 'm:row';
import 'm:size=16|20|24|28|32|40|56';

export default decl(
  {
    block: 'Header',
    ...
  }
);
```

was previously turned into

<img width="750" alt="before" src="https://user-images.githubusercontent.com/12268898/37401772-3b31e6f6-279a-11e8-940c-e0f9f44794b1.png">

and now it's

<img width="750" alt="after" src="https://user-images.githubusercontent.com/12268898/37401827-751f5b78-279a-11e8-9648-810f3fbfa3c7.png">

where requiring of _Header.js_ from itself results in an object:
(kind of webpack's way of handling recursive require?)

<img width="187" alt="require result" src="https://user-images.githubusercontent.com/12268898/37402055-2c5e5dac-279b-11e8-8e8a-005716622de0.png">
(caught on a breakpoint, and "102" is a webpack's module id of a _Header.js_ script)


So on the 25th line above there is an error of "applyDecls is not a function", because it's, well, undefined.

**Comment by keann** [comment]
Thanks a lot for the latest fixes, with 0.6.2 `require`-chains look much better! For the one who might read this thread in future: as a result the loading order became 0 -> 1 -> 2, what deprives undocumented, yet convenient possibility to make rules in `1` the most important (w/o `!important` or specifity tricks). But, according to indices, it's the most obvious order, and I totally accept it.

I'm afraid, there is another problem now 😅 [Here](https://github.com/bem/webpack-bem-loader/pull/70/files#diff-168726dbe96b3ce427e7fedce31bb0bcR67) context of a block seems to be its folder. If we have inside this folder some related test- or preview-files, that condition also prevents from pushing a `require` of a block itself inside them. As a result it's impossible to import a block anywhere in its folder.
